src/System.CommandLine/Parsing/StringExtensions.cs

```diff
@@ -579,7 +580,7 @@ IEnumerable<string> SplitLine(string line)
 
                     for (var childAliasIndex = 0; childAliasIndex < child.RawAliases.Count; childAliasIndex++)
                     {
-                        var childAlias = child.RawAliases[childAliasIndex];
+                        var childAlias = child.RawAliases.ToArray()[childAliasIndex];
```

Comment:
Any reason to not use ElementAt here?
```suggestion
                        var childAlias = child.RawAliases.ElementAt(childAliasIndex);
```